.github/workflows/plugins_react_tests.yml

```
@@ -60,17 +64,29 @@ jobs:
       - name: Install Foreman npm dependencies
         run: npm ci --no-audit
         working-directory: ${{ github.workspace }}/projects/foreman
-      - name: Checkout ${{ matrix.plugin }}
+      - name: Checkout ${{ matrix.plugin.repo }}
         uses: actions/checkout@v4
         with:
-          repository: ${{ matrix.plugin }}
+          repository: ${{ matrix.plugin.repo }}
           path: ${{ github.workspace }}/projects/plugin
-      - name: Generate ${{ matrix.plugin }} npm dependencies package-lock
+      - name: Set up plugin in Foreman
+        run: |
+          echo "gemspec name: '${{ matrix.plugin.name }}', path: '${{ github.workspace }}/projects/plugin'" > "bundler.d/${{ matrix.plugin.name }}.local.rb"
+          if [ -d ${{ matrix.plugin.name }}/gemfile.d ] ; then
+            cat ${{ matrix.plugin.name }}/gemfile.d/*.rb >> bundler.d/${{ matrix.plugin.name }}.local.rb
+          fi
+        working-directory: ${{ github.workspace }}/projects/foreman
+      - name: Generate ${{ matrix.plugin.repo }} npm dependencies package-lock
         run: npm install --package-lock-only --no-audit --legacy-peer-deps
         working-directory: ${{ github.workspace }}/projects/plugin
-      - name: Install ${{ matrix.plugin }} npm dependencies
+      - name: Install ${{ matrix.plugin.repo }} npm dependencies
```

```suggestion
      - name: Install ${{ matrix.plugin.name }} npm dependencies
```
same, nicer output :)